impl/src/main/java/org/jboss/cdi/tck/tests/beanContainer/assignability/BeanEventAssignabilityTest.java

```diff
@@ -0,0 +1,252 @@
+    public void testEventMatchingParameterized() {
+        TypeLiteral<List<String>> listOfString = new TypeLiteral<>() {};
+        TypeLiteral<List<?>> listOfWildcard = new TypeLiteral<>() {};
+        TypeLiteral<List<Integer>> listOfInteger = new TypeLiteral<>() {};
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(listOfString.getType(), Set.of(), listOfString.getType(), Set.of()),
+                "Event with Parameterized type should match when same Parameterized type is requested");
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(listOfString.getType(), Set.of(), listOfWildcard.getType(), Set.of()),
+                "Event with Parameterized type should match when Parameterized type with wildcard is requested");
+        assertFalse(getCurrentBeanContainer().isMatchingEvent(listOfString.getType(), Set.of(), listOfInteger.getType(), Set.of()),
+                "Event with Parameterized type should not match when a different Parameterized type is requested");
+    }
+
+    @Test
+    @SpecAssertion(section = BM_BEAN_EVENT_ASSIGNABILITY, id = "bb")
+    public void testEventMatchingAnyQualifier() {
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEvent.class, Set.of(Any.Literal.INSTANCE)),
+                "Event with no qualifiers should match when @Any required");
+        
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE), MyEvent.class, Set.of(Any.Literal.INSTANCE)),
+                "Event with @Qualifier1 should match when @Any required");
+    }
+
+    @Test
+    @SpecAssertion(section = BM_BEAN_EVENT_ASSIGNABILITY, id = "bc")
+    public void testEventMatchingNullException() {
+        assertThrows("Null event type", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingEvent(null, Set.of(), MyBean.class, Set.of()));
+        assertThrows("Null event qualifiers", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingEvent(MyEvent.class, null, MyEvent.class, Set.of()));
+        assertThrows("Null required type", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), null, Set.of()));
+        assertThrows("Null required qualifiers", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEvent.class, null));
+    }
+
+    @Test
+    @SpecAssertion(section = BM_BEAN_EVENT_ASSIGNABILITY, id = "bd")
+    public <X> void testEventMatchingTypeVarException() {
+        TypeLiteral<List<X>> varEventType = new TypeLiteral<>() {};
+
+        assertThrows("Type variable in event type", IllegalArgumentException.class,
+                () -> getCurrentBeanContainer().isMatchingEvent(varEventType.getType(), Set.of(), MyBean.class, Set.of()));
+
+        assertThrows("Type variable in observed event type", IllegalArgumentException.class,
```

Comment:
@Azquelt I might be missing something obvious but which part of specification does this test? The linked section is only for the event (payload) type. I don't see a similar restriction for the observed type?

Reply:
You're quite right, that's a mistake. I've removed that check.